### DISCUSSION TITLE
DJ interface doesn't work without hot plugging

### DISCUSSION THREAD

**Discussion by RK11111111111** [fayaaz/mixxx-pi-gen#85] [General]
My DJ interface doesn't work on boot.  I have to unplug then hot plug it.  I have tried to troubleshoot it.  I am wondering if anyone else has this problem.  I have tried some basics like lsusb, and it is there.  I have tried some things like triggering a rescan.  The weird thing is that lsusb says it is there, but the device, DJ Hercules Air, doesn't do the button flash thing that it usually does when it starts.  This may just be an issue with the Hercules Air, seems like I had similar issues with the Mac based player I was using, having to hotplug it often, though it seems like the DJ software by Mixvibes would trigger the interface lights flashing boot up.  I wonder if Mixxx has some functionality to initialize a DJ interface.

**Comment by jameswilliamknight** [comment]
I seemed to have this issue in a version about 1-2 months ago, my pioneer DDJ-400 required reselecting/reconfiguration in Mixxx each boot

**Reply by fayaaz** [reply]
Hmmm, this is odd. Could it be that your setting were not saved for some reason? Could you tell me which version you were using (of Mixxx and/or the image)?

**Reply by jameswilliamknight** [reply]
It had to be v0.9.1 because I just missed the boat for the 2.4 stable release

**Comment by fayaaz** [comment]
I get a slightly related one with my Korg Kaoss DJ I have to quit Mixxx, replug it in and start Mixxx. Not sure what the answer is.

**Reply by RK11111111111** [reply]
It was weird right after posting I went to my music room fired up all of my pi and it worked first time.  Shut it down and tried again no luck.

I think it might have something to do with not shutting down correctly.  That is one hypothesis, that the fdisk check at the beginning messes up the boot.

**Comment by RK11111111111** [comment]
I think it is a power issue.  I purchased the canna kit power supply 3.5 amp as compared to the phone chargers 1.5-2amp,and it has consistently started the usb on boot.  Though IMO it could also be considered bug, in the boot sequence.
